gallery/how_to/work_with_relay/using_with_pipeline_executor.py

```diff
@@ -0,0 +1,262 @@
+# ---------------------------------------------------------
+# Define a function to do all the codegen and pipeline executor works.
+# To run pipeline executor with dnnl, USE_PIPELINE_EXECUTOR need to get set as ON.
+# and the 'USE_CUTLASS' should set as ON in config.cmake.
+from tvm.contrib import graph_executor, pipeline_executor, pipeline_executor_build
+
+#########################################
+# Create subgraph pipeline configuration.
+# Associate the subgraph module with a target.
+# Using BYOC to set the codegen of the second subgraph module.
+# To use cutlass the 'USE_CUTLASS' should set as ON.
+mod0, mod1 = subgraphs[0], subgraphs[1]
+# Use cutlass as the codegen.
+mod1 = partition_for_cutlass(mod1)
+#################################################
+# Get the pipeline executor configuration object.
+pipe_config = pipeline_executor_build.PipelineConfig()
+###########################################################################
+# Set the compile target of the second subgraph module for example as LLVM.
+pipe_config[mod0].target = "llvm"
+pipe_config[mod0].dev = tvm.cpu(0)
+###############################################################################
+# Set the cpu afinity for control flow, for example using cpu 0 for control flow.
+pipe_config[mod1].cpu_affinity = "0"
+pipe_config[mod1].export_cc = None
+##############################################################
+# Set the compile target of the second subgraph module as LLVM.
+pipe_config[mod1].target = "cuda"
+pipe_config[mod1].dev = tvm.device("cuda", 0)
+pipe_config[mod1].build_func = cutlass_build
+pipe_config[mod1].export_cc = "nvcc"
+#################################################################################
+# Set the cpu afinity for control flow, for example using cpu 1 for control flow.
+pipe_config[mod1].cpu_affinity = "1"
+pipe_config["input"]["data"].connect(pipe_config[mod0]["input"]["data"])
+pipe_config[mod0]["output"][0].connect(pipe_config[mod1]["input"]["data_n_0"])
+pipe_config[mod1]["output"]["0"].connect(pipe_config["output"][0])
+######################################
+# The pipeline configuration as below.
+"""
+print(pipe_config)
+ Inputs
+  |data: mod0:data
+
+ output
+  |output(0) : mod1.output(0)
+
+ connections
+  |mod0.output(0)-> mod1.data_n_0
+"""
+
+# sphinx_gallery_start_ignore
+from tvm import testing
+
+# testing.utils.install_request_hook(depth=3)
+# sphinx_gallery_end_ignore
+##############################
+# Build the pipeline executor.
+# ----------------------------
+with tvm.transform.PassContext(opt_level=3):
+    pipeline_mod_factory = pipeline_executor_build.build(pipe_config)
+###############################################
+# Export the parameter configuration to a file.
+directory_path = tvm.contrib.utils.tempdir().temp_dir
+#############################################
+# If the directory does not exist, create it.
+if not os.path.exists(directory_path):
+    os.makedirs(directory_path)
```

Comment:
Just use `makedirs(directory_path, exist_ok=True)` and remove `if not os.path.exists(directory_path):` and the sentence above it.

Reply:
fixed.